client.go

```diff
@@ -187,7 +196,15 @@ func NewClient(options ...Option) *Client {
 	case opts.skipHostnameVerification:
 		transport = transportWithSkipVerify(transport, opts.skipHostnameVerification)
 	}
-	client.Transport = transport
+
+	if opts.requestRecorder != nil {
+		client.Transport = roundTripRecorder{
```

Comment:
The middleware work with *http.Transport values whereas in this case we just need to provide a custom RoundTripper. The only solution was to replace client.Transport (which is a RoundTripper instance)